packages/js/product-editor/src/components/text-control/text-control.tsx

```diff
@@ -46,7 +46,11 @@ export const TextControl = forwardRef( function ForwardedTextControl(
 			}
 			required={ required }
 			help={ error || help }
-			onChange={ onChange }
+			onChange={ ( value: string | undefined ) => {
+				if ( value && onChange ) {
+					onChange( value );
+				}
+			} }
```

Comment:
```suggestion
			onChange={ ( value: string | undefined ) => {
				if ( onChange ) {
					onChange( value ?? '' );
				}
			} }
```
Same as password field. Also its weird to me that the `onChange` is optional here. Should we just make it required? or will that cause a entire other set of TS errors?

Reply:
Fixed with e1f833ffa51d6d0e7100adcce0347930fee5be5f